spec/classes/config_spec.rb

```diff
@@ -77,6 +77,18 @@
     }
   end
 
+  context 'log4j file' do
+    log4j_file = '/tmp/log4j.properties'
+    File.open(log4j_file, "w") { |file| file.write("log4j.rootLogger=INFO, CONSOLE") }
```

Comment:
Puppet rspec tests do not actually deploy files, they just inspect what "would" happen by looking at the catalog.

Just inject params and assert that it should contain_file with that line. And maybe also that it should `contain_file(mainlog4jfile).with_link(log4j_file)`

No need to do file io at all, again the tests here just inspect the catalog.